### PR TITLE
Fix config-settings handling in pip install

### PR DESCRIPTION
**Pull request by hcho3**
Also test `--config-settings` in the CI so that we know it is not broken.

### CODE REVIEW

**Comment by hcho3** [comment]
```
Collecting pyspark (from -r doc/requirements.txt (line 13))
  Downloading pyspark-3.4.0.tar.gz (310.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 310.8/310.8 MB 174.2 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      ERROR: Can not execute `setup.py` since setuptools is not available in the build environment.
      [end of output]  
```
For some reason, the doc build is broken :(

File: python-package/packager/pep517.py [hunk 1]
```diff
@@ -79,7 +79,8 @@ def build_wheel(
         libxgboost = locate_or_build_libxgboost(
             TOPLEVEL_DIR, build_dir=build_dir, build_config=build_config
         )
-        copy_with_logging(libxgboost, lib_path, logger=logger)
+        if not build_config.use_system_libxgboost:
+            copy_with_logging(libxgboost, lib_path, logger=logger)
```
Comment (hcho3) [inline python-package/packager/pep517.py:83]: I manually tested this change by:
1. Create a new Conda environment.
2. Build XGBoost from the source using CMake:
```
conda activate my_env
cmake .. -GNinja -DCMAKE_INSTALL_PREFIX="$CONDA_PREFIX" -DCMAKE_INSTALL_LIBDIR="lib"
```
3. Install libxgboost into the Conda env:
```
ninja install
```
4. Install the Python package with `use_system_libxgboost` flag:
```
pip install -v .  --config-settings use_system_libxgboost=True
```
5. Inspect `$CONDA_PREFIX` to ensure that only one copy of `libxgboost.so` is installed in the Conda environment.

**Comment by hcho3** [comment]
I filed https://github.com/readthedocs/readthedocs.org/issues/10285 to notify RTD about the broken doc build.

**Comment by trivialfis** [comment]
Yup, let's skip the unrelated errors.

@rongou Have you seen this error before https://buildkite.com/xgboost/xgboost-ci/builds/2386 ?

**Comment by rongou** [comment]
@trivialfis no, but will take a look.

**Review by trivialfis — APPROVED** [review]
Feel free to merge it.

**Pull request merged**